packages/dashboard/src/components/sidePanel/sections/baseSettingSection/index.spec.tsx

```diff
@@ -32,9 +39,28 @@ it('renders', () => {
 });
 
 it('renders all input fields', () => {
-  const elem = render(<TestBaseSettingSection />).baseElement;
-  expect(elem.querySelector('[data-test-id="base-setting-x-input"]')).toBeTruthy();
-  expect(elem.querySelector('[data-test-id="base-setting-y-input"]')).toBeTruthy();
-  expect(elem.querySelector('[data-test-id="base-setting-height-input"]')).toBeTruthy();
-  expect(elem.querySelector('[data-test-id="base-setting-width-input"]')).toBeTruthy();
+  const wrapper = createWrapper(render(<TestBaseSettingSection />).baseElement);
+  expect(wrapper.findInput('[data-test-id="base-setting-x-input"]')).toBeTruthy();
+  expect(wrapper.findInput('[data-test-id="base-setting-y-input"]')).toBeTruthy();
+  expect(wrapper.findInput('[data-test-id="base-setting-width-input"]')).toBeTruthy();
+  expect(wrapper.findInput('[data-test-id="base-setting-height-input"]')).toBeTruthy();
+});
+
+it('rounds x,y,height,width input', () => {
+  const wrapper = createWrapper(render(<TestBaseSettingSection />).baseElement);
+  expect(wrapper.findInput('[data-test-id="base-setting-x-input"]')?.findNativeInput().getElement().value).toBe('2');
+  expect(wrapper.findInput('[data-test-id="base-setting-y-input"]')?.findNativeInput().getElement().value).toBe('2');
+  expect(wrapper.findInput('[data-test-id="base-setting-width-input"]')?.findNativeInput().getElement().value).toBe(
+    '10'
+  );
+  expect(wrapper.findInput('[data-test-id="base-setting-height-input"]')?.findNativeInput().getElement().value).toBe(
+    '11'
+  );
+});
+
+it('rounds input on user input', async () => {
+  const wrapper = createWrapper(render(<TestBaseSettingSection />).baseElement);
+  const xInput = wrapper.findInput('[data-test-id="base-setting-x-input"]')!.findNativeInput().getElement();
+  await userEvent.type(xInput!, '2.4'); // current x is 2, this type appends 2 after it.
+  expect(xInput?.value).toBe('22'); // decimal digits on typing will be ignored not rounded.
```

Comment:
shouldn't this be just 2? or would it end up being 24?

Reply:
Current X for the widget is `2` and `type` event just append new chars after it.  So it is `2` -> `22.4` -> `22`

Reply:
Oh I see, so the idea is you're attempting to change the x from 2 -> 22.4

I thought it was a scenario trying to change 2 -> 2.4

perfect lgtm